### DISCUSSION TITLE
Pressure sensor(atmosphere)

### DISCUSSION THREAD

**Discussion by danilkorotkov** [HomeSpan/HomeSpan#254] [Q&A] [answered]
Is this sensor absent in HAP?

**Comment by HomeSpan** [comment]
HAP-R2 does not include a pressure sensor, though if you are aware of it being supported in the Home App (similar to TV Controls) you can add it in using the Custom Characteristic macros.   You can also also use Custom Characteristics for adding new functionality that is recognized by other applications, such as Eve for HoneKit.

**Reply by danilkorotkov** [reply]
Have you any description how to use custom?

**Comment by danilkorotkov** [comment]
is it possible with homespan?
https://github.com/rxseger/homebridge-bme280/blob/master/lib/CustomCharacteristic.js#L12?ref=https://githubhelp.com
https://gist.github.com/simont77/3f4d4330fa55b83f8ca96388d9004e7d?permalink_comment_id=3804953
```  
CustomCharacteristic.AtmosphericPressureLevel = function() {
    Characteristic.call(this, 'Air Pressure', CustomCharacteristic.AtmosphericPressureLevel.UUID);
    this.setProps({
      format: Characteristic.Formats.UINT8,
      unit: "mbar",
      minValue: 800,
      maxValue: 1200,
      minStep: 1,
      perms: [Characteristic.Perms.READ, Characteristic.Perms.NOTIFY]
    });
    this.value = this.getDefaultValue();
  };
  CustomCharacteristic.AtmosphericPressureLevel.UUID = 'E863F10F-079E-48FF-8F27-9C2605A29F52';
  inherits(CustomCharacteristic.AtmosphericPressureLevel, Characteristic);


  // courtesy of https://github.com/robi-van-kinobi/homebridge-cubesensors
  CustomCharacteristic.AtmosphericPressureSensor = function(displayName, subtype) {
    Service.call(this, displayName, CustomCharacteristic.AtmosphericPressureSensor.UUID, subtype);

    // Required Characteristics
    this.addCharacteristic(CustomCharacteristic.AtmosphericPressureLevel);

    // Optional Characteristics
    this.addOptionalCharacteristic(Characteristic.StatusActive);
    this.addOptionalCharacteristic(Characteristic.StatusFault);
    this.addOptionalCharacteristic(Characteristic.StatusLowBattery);
    this.addOptionalCharacteristic(Characteristic.StatusTampered);
    this.addOptionalCharacteristic(Characteristic.Name);
  };
  CustomCharacteristic.AtmosphericPressureSensor.UUID = 'B77831FD-D66A-46A4-B66D-FD7EE8DFE3CE';
  inherits(CustomCharacteristic.AtmosphericPressureSensor, Service);```

**Comment by danilkorotkov** [comment, marked as answer]
done https://github.com/danilkorotkov/GY-21P-HomeKit
barometric pressure only in Eve app

**Comment by Xorlent** [comment]
I did create a virtual HomeKit switch that monitored a pressure sensor.  If rapid vacuum was detected, it would shut off the virtual switch.  Basically it creates a safety in case a house fan is turned on without enough windows open.  Obviously the pressure value is only maintained on the microcontroller and not seen directly in HomeKit.

**Reply by danilkorotkov** [reply]
Eve supports pressure and scenes
